src/lv_objx/lv_chart.c

```diff
@@ -1046,65 +1096,187 @@ static void lv_chart_draw_areas(lv_obj_t * chart, const lv_area_t * mask)
+/**
+ * Create iterator for newline-separated list
+ * @param list pointer to newline-separated labels list
+ * @param iterator_dir LV_CHART_ITERATOR_FORWARD or LV_CHART_LABEL_ITERATOR_REVERSE
+ * @return label_iterator_t
+ */
+static label_iterator_t lv_chart_create_label_iter(const char * list, uint8_t iterator_dir)
+{
+    label_iterator_t iterator = {0};
+    uint8_t j;
+
+    iterator.list_start = list;
+
+    /* count number of list items */
+    for(j = 0; list[j] != '\0'; j++) {
+            if(list[j] == '\n')
+               iterator.items_left++;
+    }
+
+    if(iterator_dir == LV_CHART_LABEL_ITERATOR_FORWARD) {
+        iterator.is_reverse_iter = 0;
+        iterator.current_pos = list;
+    } else {
+        iterator.is_reverse_iter = 1;
+        // -1 to skip '\0' at the end of the string
+        iterator.current_pos = list + j - 1;
+    }
+    iterator.items_left++;
+    return iterator;
+}
+
+/**
+ * Get next label from iterator created by lv_chart_create_label_iter()
+ * @param iterator iterator to get label from
+ * @param[out] buf buffer to point next label to
+ */
+static void lv_chart_get_next_label(label_iterator_t * iterator, char * buf)
+{
+    uint8_t label_len = 0;
+    if (iterator->is_reverse_iter) {
+        const char * label_start;
+        /* count the length of the current label*/
+        while ((*iterator->current_pos != '\n') &&
+                (iterator->current_pos != iterator->list_start)) {
+            iterator->current_pos--;
+            label_len++;
+        }
+
+        label_start = iterator->current_pos;
+
+        if (*iterator->current_pos == '\n') {
+            /* do not copy \n symbol, +1 to skip it*/
+            label_start++;
+            /* skip newline*/
+            iterator->current_pos--;
+        } else {
+            /* it is last label in list (first one from the beginning )*/
+            label_len++;
+        }
+
+        /* do not allow output buffer overflow */
+        if (label_len > LV_CHART_AXIS_TICK_LABEL_MAX_LEN) {
+            label_len = LV_CHART_AXIS_TICK_LABEL_MAX_LEN;
+        }
+
+        strncpy(buf, label_start, label_len);
+    } else {
+        /* search for tick string */
+        while(iterator->current_pos[label_len] != '\n' &&
+            iterator->current_pos[label_len] != '\0') {
+            /* do not overflow the buffer, but move to the end of the current label */
+            if(label_len < LV_CHART_AXIS_TICK_LABEL_MAX_LEN) {
+                buf[label_len] = iterator->current_pos[label_len];
+                label_len++;
+            } else {
+                label_len++;
+            }
+        }
+
+        iterator->current_pos += label_len;
+
+        /* do not allow output buffer overflow */
+        if (label_len > LV_CHART_AXIS_TICK_LABEL_MAX_LEN) {
+            label_len = LV_CHART_AXIS_TICK_LABEL_MAX_LEN;
+        }
+
+        if(*(iterator->current_pos) == '\n') iterator->current_pos++;
```

Comment:
Nitpicking here - any reason why you put parentheses around `iterator->current_pos` here but not in the other `if` statements?

Reply:
My bad. If you tell me what is the preferred way I'll make them all look uniform.

Reply:
The coding style actually doesn't specify one way or the other. (I just checked.)

In that case just remove the parentheses; I don't think they are used that way anywhere else in the tree.

Reply:
FIxed by 9fbad74